source/base/data_out_base.cc

```diff
@@ -89,6 +89,13 @@ namespace
 
 namespace
 {
+#ifdef DEAL_II_WITH_ZLIB
+  constexpr bool deal_ii_with_zlib = true;
+#else
+  constexpr bool deal_ii_with_zlib = false;
+#endif
```

Comment:
once we require C++17 we can do this properly!

Reply:
How would you do this with C++17? I'm translating a preprocessor symbol into a compiler variable, and I don't know a different way to do that.

There are places further down where we can use `if constexpr`, though.

Reply:
We wouldn't change this, but with C++17 we can replace a lot of preprocessor blocks (like the ones in this PR) with `if constexpr`.

Reply:
Yes, true. In fact, we could change a lot of things in `config.h` where we currently `#define` something into `constexpr` something. I, for one, look forward to that!